hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -179,6 +180,9 @@ private Schema generateRequiredSchema() {
     for (String field : recordMerger.getMandatoryFieldsForMerging(hoodieTableConfig)) {
       if (requestedSchema.getField(field) == null) {
         Option<Schema.Field> foundFieldOpt  = findNestedField(dataSchema, field);
+        if (!foundFieldOpt.isPresent()) {
+          foundFieldOpt = findMetaField(field);
```

Comment:
Why the record key meta field is a supplanted fix?

Reply:
Do you mean why we fixed it here? Because when I build the test framework, and some test cases can not pass.

Reply:
Can you amplify the details, how the test got fixed with this change? I want to understand why the `_hoodie_record_key` is necessary here?

Reply:
I think it's part of the record merger contract. One or more are needed to decide the merge. It need not necessarily be `_hoodie_record_key`. See https://github.com/apache/hudi/blob/50f0d9f3baeafe92c38dfb0b59b337d83391ea42/hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordMerger.java#L129